src/libraries/System.Private.CoreLib/src/System/Environment.SunOS.cs

```diff
@@ -7,7 +7,6 @@ namespace System
 {
     public static partial class Environment
     {
-        public static long WorkingSet =>
-            (long)(Interop.procfs.TryReadProcessStatusInfo(ProcessId, out Interop.procfs.ProcessStatusInfo status) ? status.ResidentSetSize : 0);
+        public static long WorkingSet => (long)(Interop.procfs.TryGetProcessInfoById(ProcessId, out Interop.procfs.ProcessInfo iProcInfo) ? iProcInfo.ResidentSetSize : 0);
```

Comment:
```
 public static long WorkingSet => Interop.procfs.TryGetProcessInfoById(ProcessId, out Interop.procfs.psinfo psinfo) ? psInfo.ResidentSetSize : 0;
```
This (and other similar methods) look inefficient:
1. We read psinfo info first
2. Then convert it to ProcessInfo, creating some strings in the process
3. Then throw away most of it and just fetch a single integer

Should TryGetProcessInfoById return psinfo directly to allow callers only fetch what they need?

Reply:
This is improved now.  Please have another look.